app/internal/upstream/upstream_test.go

```
@@ -911,47 +1095,409 @@ func TestSpecsFromDockerEnv(t *testing.T) {
+	if err != nil {
+		t.Fatalf("NewRequest: %v", err)
 	}
-	if ep.TLSConfig.RootCAs != nil {
-		t.Error("RootCAs is non-nil, want nil (use system roots)")
+	conn, upstreamReq, err := r.DialRequest(context.Background(), req)
+	if err != nil {
+		t.Fatalf("DialRequest: %v", err)
 	}
-	if len(ep.TLSConfig.Certificates) != 0 {
-		t.Error("client certificate present, want none (server-auth only)")
+	defer conn.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `conn.Close` is not checked

(errcheck)